### PR TITLE
GenericLoad() done

### PR DESCRIPTION
**Pull request by whampson**
Finished `GenericLoad()`

### CODE REVIEW

File: src/save/GenericGameStorage.cpp [hunk 1]
```diff
@@ -198,6 +210,104 @@ GenericSave(int file)
...
+	ReadDataFromBufferPointer(buf, CClock::ms_nLastClockTick);
+	ReadDataFromBufferPointer(buf, CClock::ms_nGameClockHours);
+	ReadDataFromBufferPointer(buf, CClock::ms_nGameClockMinutes);
+	currPad = CPad::GetPad(0);
+	ReadDataFromBufferPointer(buf, currPad->Mode);
+	ReadDataFromBufferPointer(buf, timeInMilliseconds); CTimer::SetTimeInMilliseconds(timeInMilliseconds);
+	ReadDataFromBufferPointer(buf, timeScale); CTimer::SetTimeScale(timeScale);
+	ReadDataFromBufferPointer(buf, timeStep); CTimer::SetTimeStep(timeStep);
+	ReadDataFromBufferPointer(buf, timeStepNonClipped); CTimer::SetTimeInMillisecondsNonClipped(timeStepNonClipped);
+	ReadDataFromBufferPointer(buf, frameCounter); CTimer::SetFrameCounter(frameCounter);
```
Comment (Sergeanur) [inline src/save/GenericGameStorage.cpp:253]: Original code writes straight to CTimer fields. Mayble make GenericLoad a friendly func?

File: src/save/GenericGameStorage.cpp [hunk 2]
```diff
@@ -58,10 +59,22 @@ uint32 &TimeToStayFadedBeforeFadeOut = *(uint32*)0x611564;
...
-
+#define LoadSaveDataBlock(msg,load_func)\
+do {\
+	if (!ReadDataFromFile(file, (uint8 *) &size, 4))\
+		return false;\
+	size = align4bytes(size);\
+	if (!ReadDataFromFile(file, work_buff, size))\
+		return false;\
+	buf = work_buff;\
+	printf(msg);\
```
Comment (erorcun) [inline src/save/GenericGameStorage.cpp:70]: debug()

File: src/save/GenericGameStorage.cpp [hunk 3]
```diff
@@ -198,6 +210,104 @@ GenericSave(int file)
...
+	printf("Loading Scripts \n");
+	ReadDataFromBufferPointer(buf, size);
+	CTheScripts::LoadAllScripts(buf, size);
+	size = align4bytes(size);
+	buf += size;
+
+	// Load the rest
+	LoadSaveDataBlock("Loading PedPool \n", CPools::LoadPedPool);
+	LoadSaveDataBlock("Loading Garages \n", CGarages::Load);
+	LoadSaveDataBlock("Loading Vehicles \n", CPools::LoadVehiclePool);
```
Comment (erorcun) [inline src/save/GenericGameStorage.cpp:280]: you're missing
        CProjectileInfo::RemoveAllProjectiles();
        CObject::DeleteAllTempObjects();
here

**Comment by erorcun** [comment]
lgtm except comments

File: src/save/GenericGameStorage.cpp (hunk 1 above)
Comment (whampson) [reply src/save/GenericGameStorage.cpp]: Made `GenericLoad` and `GenericSave` friends to `CTimer` and wrote the fields directly in both functions.

File: src/save/GenericGameStorage.cpp (hunk 3 above)
Comment (whampson) [reply src/save/GenericGameStorage.cpp]: Good catch.
I had to split up the macro because of where these functions are called.

**Pull request merged**